snippets/core/system-text-json/csharp/ImmutablePointConverter.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class ImmutablePointConverter : JsonConverter<ImmutablePoint>
```

Comment:
I don't know if this is the right place for this, but this converter can be written more efficiently. So, maybe we suggest that?

1) `reader.GetString()` for property name matching isn't necessary. You could use `ValueTextEquals`
2) Comparison with `JsonEncodedText` is faster than comparison with `string`.
3) Rather than asking for an int converter every time, it's probably better to get it once and either pass it in, or store it as a field by getting it in the custom ctor. cc @steveharter, @layomia 
4) You can remove one of the checks at the end (`if (!xSet || !ySet)`) by crafting the conditions differently.

```C#
public class ImmutablePointNewConverter : JsonConverter<ImmutablePoint>
{
    private readonly JsonEncodedText XName = JsonEncodedText.Encode("X");
    private readonly JsonEncodedText YName = JsonEncodedText.Encode("Y");

    private readonly JsonConverter<int> _intConverter;

    public ImmutablePointNewConverter(JsonSerializerOptions options)
    {
        if (options?.GetConverter(typeof(int)) is JsonConverter<int> intConverter)
        {
            _intConverter = intConverter;
        }
        else
        {
            throw new InvalidOperationException();
        }
    }

    public override ImmutablePoint Read(
        ref Utf8JsonReader reader,
        Type typeToConvert,
        JsonSerializerOptions options)
    {
        if (reader.TokenType != JsonTokenType.StartObject)
        {
            throw new JsonException();
        };

        int x = default;
        bool xSet = false;

        int y = default;
        bool ySet = false;

        // Get the first property.
        reader.Read();
        if (reader.TokenType != JsonTokenType.PropertyName)
        {
            throw new JsonException();
        }

        if (reader.ValueTextEquals(XName.EncodedUtf8Bytes))
        {
            x = ReadProperty(ref reader, options);
            xSet = true;
        }
        else if (reader.ValueTextEquals(YName.EncodedUtf8Bytes))
        {
            y = ReadProperty(ref reader, options);
            ySet = true;
        }
        else
        {
            throw new JsonException();
        }

        // Get the second property.
        reader.Read();
        if (reader.TokenType != JsonTokenType.PropertyName)
        {
            throw new JsonException();
        }

        if (xSet && reader.ValueTextEquals(YName.EncodedUtf8Bytes))
        {
            y = ReadProperty(ref reader, options);
        }
        else if (ySet && reader.ValueTextEquals(XName.EncodedUtf8Bytes))
        {
            x = ReadProperty(ref reader, options);
        }
        else
        {
            throw new JsonException();
        }

        reader.Read();

        if (reader.TokenType != JsonTokenType.EndObject)
        {
            throw new JsonException();
        }

        return new ImmutablePoint(x, y);
    }

    private int ReadProperty(ref Utf8JsonReader reader, JsonSerializerOptions options)
    {
        Debug.Assert(reader.TokenType == JsonTokenType.PropertyName);

        reader.Read();
        return _intConverter.Read(ref reader, typeof(int), options);
    }

    public override void Write(
        Utf8JsonWriter writer,
        ImmutablePoint value,
        JsonSerializerOptions options)
    {
        // Don't pass in options when recursively calling Serialize.
        JsonSerializer.Serialize(writer, value);
    }
}
```